src/libPMacc/include/cuSTL/container/CartBuffer.tpp

```diff
@@ -198,7 +199,7 @@ CartBuffer<Type, T_dim, Allocator, Copier, Assigner>::operator=
             std::stringstream() << "Assignment: Sizes of buffers do not match: "
                 << this->size() << " <-> " << rhs.size() << std::endl).str());
 #else
-    assert(rhs.size() == this->size());
+    PMACC_ASSERT(rhs.size() == this->size());
```

Comment:
this create a WARNINGS: `warning: calling a __host__ function from a __host__ __device__ function is not allowed
`